core/vibes/soul/primitives/compare-drawer/index.tsx

```diff
@@ -0,0 +1,207 @@
+            <div className="flex flex-1 flex-wrap justify-end gap-4">
+              {optimisticItems.map((item) => (
+                <div className="relative" key={item.id}>
+                  <Link
+                    className="group relative flex max-w-56 items-center overflow-hidden whitespace-nowrap rounded-xl border border-[var(--compare-drawer-link-border,hsl(var(--contrast-100)))] bg-[var(--compare-drawer-card-background,hsl(var(--background)))] font-semibold ring-[var(--compare-drawer-card-focus,hsl(var(--primary)))] transition-all duration-150 hover:bg-[var(--compare-drawer-card-background-hover,hsl(var(--contrast-100)))] focus:outline-none focus:ring-2"
+                    href={item.href}
+                  >
+                    <div className="relative aspect-square w-12 shrink-0 bg-[var(--compare-drawer-card-image-background,hsl(var(--contrast-100)))]">
+                      {item.image?.src != null ? (
+                        <Image
+                          alt={item.image.alt}
+                          className="rounded-lg object-cover @4xl:rounded-r-none"
+                          fill
+                          sizes="3rem"
+                          src={item.image.src}
+                        />
+                      ) : (
+                        <span className="max-w-full break-all p-1 text-xs text-[var(--compare-drawer-empty-image-text,color-mix(in_oklab,hsl(var(--primary)),black_75%))] opacity-20">
+                          {getInitials(item.title)}
+                        </span>
+                      )}
+                    </div>
+                    <span className="hidden truncate pl-3 pr-5 text-[var(--compare-drawer-card-text,hsl(var(--foreground)))] @4xl:block">
+                      {item.title}
+                    </span>
+                  </Link>
+                  <button
+                    aria-label={`Remove ${item.title}`}
+                    className="hover:text-[var(--compare-drawer-dismiss-icon-hover,hsl(var(--foreground))] absolute -right-2.5 -top-2.5 flex h-7 w-7 items-center justify-center rounded-full border border-[var(--compare-drawer-dismiss-border,hsl(var(--contrast-100)))] bg-[var(--compare-drawer-dismiss-background,hsl(var(--background)))] text-[var(--compare-drawer-dismiss-icon,hsl(var(--contrast-400)))] transition-colors duration-150 hover:border-[var(--compare-drawer-dismiss-border-hover,hsl(var(--contrast-200)))] hover:bg-[var(--compare-drawer-dismiss-background-hover,hsl(var(--contrast-100)))]"
+                    onClick={() => {
+                      startTransition(async () => {
+                        await setParam((prev) => {
+                          const next = prev?.filter((v) => v !== item.id) ?? [];
+
+                          setOptimisticItems({ type: 'remove', item });
+
+                          return next.length > 0 ? next : null;
+                        });
+                      });
```

Comment:
🍹 Is there any way we could use a link here instead? If JS is disabled in the browser, then users won't be able to remove filters since this is handled by an `onClick` handler.

Reply:
Looking into it.

Reply:
This pattern reoccurs in other components. I will create a ticket for it so we can look into the best pattern.